### PR TITLE
Optimize resolve_expression to operate on a single scope vector

### PR DESCRIPTION
**Pull request by milianw**
Instead of cloning the vector on every iteration level, pass the scope in and out of the visitation function and push/pop the element there as needed. This way we can operate on a single vector that gets moved around, which removes a few thousand memory allocations. The speed impact is not measurable, as the code also triggers rowan API that is much more allocation happy.

Still, I believe this patch is still merge-worthy as it also reduces the code duplication a bit and is subjectively better, esp. from a performance pov.

### CODE REVIEW

File: internal/compiler/passes/resolving.rs [hunk 1]
```diff
@@ -76,6 +76,24 @@ fn resolve_expression(
...
 
+/// Call the visitor for each children of the element recursively, starting with the element itself
+///
+/// The item that is being visited will be pushed to the scope and popped once visitation is over.
+fn recurse_elem_with_scope(
+    elem: &ElementRc,
+    scope: ComponentScope,
+    vis: &mut impl FnMut(&ElementRc, &ComponentScope),
+) -> ComponentScope {
+    let mut scope = scope;
```
Comment (ogoffart) [inline internal/compiler/passes/resolving.rs:87]: (you can just use `mut scope` in the function declaration)

**Review by ogoffart — APPROVED** [review]
nice!

File: internal/compiler/passes/resolving.rs (hunk 1 above)
Comment (milianw) [reply internal/compiler/passes/resolving.rs]: thanks, fixed

Comment (milianw) [reply internal/compiler/passes/resolving.rs]: now actually fixed, TIL about this syntax - unexpected :)

**Pull request merged**